### PR TITLE
Update self_test

### PR DESCRIPTION
**Pull request by nabobalis**
1. Non-zero return
2. Allow `python -m sunpy.tests.self_test`

### CODE REVIEW

File: sunpy/tests/self_test.py [hunk 1]
```diff
@@ -55,10 +55,26 @@ def self_test(*, package=None, online=False, online_only=False, figure_only=Fals
...
+    parser = argparse.ArgumentParser(description="Run the sunpy test suite")
+    parser.add_argument("--package", help="The sunpy subpackage to test.")
+    parser.add_argument("--online", action="store_true",
+                        help="Run the tests that require internet access.")
+    parser.add_argument("--online-only", action="store_true",
+                        help="Run only the tests that require internet access.")
+    parser.add_argument("--figure-only", action="store_true",
+                        help="Run only the tests that generate figures.")
+    test_args = parser.parse_args()
+    self_test(package=test_args.package, online=test_args.online,
```
Comment (Cadair) [inline sunpy/tests/self_test.py:79]: This should use sys.exit.

Comment (nabobalis) [reply sunpy/tests/self_test.py]: Fixed.

**Review by Cadair — APPROVED** [review]
My only remaining comment is if it's worth running this as part of our CI?

**Review by Cadair — CHANGES_REQUESTED** [review]
Ok, so I tested this locally and it dosen't seem to work if you run it outside of the source checkout:

```
python -m "sunpy.tests.self_test"



Starting sunpy self test...
Checking for packages needed to run sunpy:
All required and optional sunpy dependencies are installed.
Starting the sunpy test suite:

==================================================================================== test session starts =====================================================================================
platform linux -- Python 3.10.8, pytest-7.1.1, pluggy-1.0.0
Matplotlib: 3.6.1
Freetype: 2.6.1
rootdir: /home/stuart
plugins: remotedata-0.3.3, openfiles-0.5.0, mock-3.7.0, filter-subpackage-0.1.1, doctestplus-0.12.0, cov-3.0.0, astropy-header-0.2.1, arraydiff-0.5.0, hypothesis-6.39.4, mpl-0.14.0, intercept-remote-1.25, asdf-2.11.0, forked-1.4.0, xdist-2.5.0, napari-plugin-engine-0.2.0, npe2-0.6.1, napari-0.4.16, astropy-0.10.0
collected 0 items                                                                                                                                                                            

=================================================================================== no tests ran in 0.47s ====================================================================================
```

**Comment by nabobalis** [comment]
Oh I think I know what it is, did you remove the `sunpy` directory from your home directory?

**Review by Cadair — APPROVED** [review]

**Pull request merged**